framework/src/wifi_manager/wifi_manager_state.c

```diff
@@ -479,8 +479,18 @@ wifi_manager_result_e _handler_on_disconnected_state(wifimgr_msg_s *msg)
 wifi_manager_result_e _handler_on_disconnecting_state(wifimgr_msg_s *msg)
 {
 	WM_LOG_HANDLER_START;
-	if (msg->event != EVT_STA_DISCONNECTED) {
+
+	if (msg->event == EVT_DEINIT_CMD) {
+		WIFIMGR_SET_SUBSTATE(WIFIMGR_DISCONN_DEINIT, msg->signal);
```

Comment:
Is it guaranteed that the deinit operation works well in any states such as the disconnecting state?
How about another deinit request during the deinit process?

Reply:
I've verified deinit operation in 5 situations init, connecting, connected, disconnecting, and scanning. you can see how did I test it at wm_test_stress_test1.c.